pkg/devices/rapl/rapl_power_meter.go

```diff
@@ -0,0 +1,38 @@
+/*
+Copyright 2025.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package rapl
+
+import (
+	"github.com/prometheus/procfs"
+	"github.com/sustainable-computing-io/kepler/pkg/devices"
+)
+
+// RAPLCPUPowerMeter implements CPUPowerMeter interface and retrieves energy
+// usage data from Intel RAPL (Running Average Power Limit) power domains using
+// prometheus/procfs package.
+//
+// It provides access to energy measurements for CPU packages, cores, and
+// other components (e.g., DRAM or uncore), depending on the actual CPU.
+//
+// Each energy zone corresponds to a powercap domain exposed by the kernel
+// through /sys/class/powercap/intel-rapl:* interfaces, as parsed by procfs.
+type RaplCPUPowerMeter struct {
+	fs *procfs.FS
+}
+
+func NewRaplCPUPowerMeter() (devices.CPUPowerMeter, error) {
```

Comment:
How about constructors just return the actual object and not the interface ?

Reply:
usually when an interface is defined, the interface is returned. Let me know if any objection to return interface.